### PR TITLE
Use a Logger for logging

### PR DESCRIPTION
**Pull request by jmccann**
Trying to standardize on log output.

Comments welcome!

### CODE REVIEW

File: lib/drone/chef/config.rb [hunk 1]
```diff
@@ -149,6 +139,17 @@ def keyfile_path
         )
       end
 
+      def logger
+        @logger ||= begin
+          l = Logger.new(STDIN)
```
Comment (tboerger) [inline lib/drone/chef/config.rb]: STDOUT

File: lib/drone/chef/config.rb [hunk 2]
```diff
@@ -58,11 +60,7 @@ def ssl_mode
       # @return [TrueClass, FalseClass]
       #
       def ssl_verify?
-        if vargs.ssl_verify.nil?
-          true
-        else
-          vargs.ssl_verify?
-        end
+        vargs.ssl_verify.nil? ? true : vargs.ssl_verify?
```
Comment (tboerger) [inline lib/drone/chef/config.rb]: I did it like that because I dislike the double question marks :D

File: lib/drone/chef/config.rb (hunk 1 above)
Comment (tboerger) [inline lib/drone/chef/config.rb]: In such cases I end up with

Logger.new(STDOUT).tap do |l|
  like.level =...
end

Than you can skip and will never forget to return the instance

**Comment by jmccann** [comment]
Thanks for the comments.  Will refactor accordingly.

**Comment by jmccann** [comment]
@tboerger How's it look now?

File: lib/drone/chef/config.rb [hunk 3]
```diff
@@ -18,8 +19,9 @@ class Config
       #
       # Initialize an instance
       #
-      def initialize(payload)
+      def initialize(payload, log = nil)
         self.payload = payload
+        @logger = log || logger
```
Comment (tboerger) [inline lib/drone/chef/config.rb]: I would define an attr_accessor like it's done for payload and set it with self.logger, than it looks pretty much the same.

**Comment by tboerger** [comment]
Tiny NITPICKing but otherwise LGTM

**Comment by kzaitsev** [comment]
LGTM

**Pull request merged**